### PR TITLE
docs: misc updates

### PR DESCRIPTION
**Pull request by jorgeorpinel**
- Clarify tracked data is needed for push/pull to start GDrive auth (per a Discord support case)
  See content/docs/user-guide/setup-google-drive-remote.md
- Update directory tracking note in add and run
  See content/docs/user-guide/dvc-files-and-directories.md,
  content/docs/command-reference/add.md, and
  content/docs/command-reference/run.md
- Updated a title and link in all other files

### CODE REVIEW

File: content/docs/command-reference/run.md [hunk 1]
```diff
@@ -73,23 +73,34 @@ $ dvc run -n printer -d write.sh -o pages ./write.sh
 $ dvc run -n scanner -d read.sh -d pages -o signed.pdf ./read.sh
 ```
 
+Stage dependencies can be any file in the workspace, either untracked, or much
```
Comment (shcheklein) [inline content/docs/command-reference/run.md:76]: mention external deps?

File: content/docs/command-reference/run.md [hunk 2]
```diff
@@ -73,23 +73,34 @@ $ dvc run -n printer -d write.sh -o pages ./write.sh
...
 - `dvc run` checks the dependency graph integrity before creating a new stage.
-  For example: two stage cannot explicitly specify the same output, there should
-  be no cycles, etc.
+  For example: two stage cannot specify the same output or overlapping output
+  paths, there should be no cycles, etc.
 
 - DVC does not feed dependency files to the command being run. The program will
   have to read by itself the files specified with `-d`.
 
+- Entire directories can be tracked as outputs, which produces a single `.dir`
```
Comment (shcheklein) [inline content/docs/command-reference/run.md:95]: produced as an output?

File: content/docs/command-reference/run.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/run.md:76]: Good point. Maybe I'll just not mention the workspace to avoid complicating the paragraph... ⌛

Comment (jorgeorpinel) [reply content/docs/command-reference/run.md]: OK removed worspace mention from here and added a bullet about external deps/outs in cb80e026.

File: content/docs/command-reference/run.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/run.md]: Yes. But "tracked" is a key concept here I think. I went for "Entire directories produced by the stage can be tracked as outputs by DVC" in c02e81e2.

**Comment by jorgeorpinel** [comment]
Thanks! Merging

**Pull request merged**